workfinder/wait/landsat8.py

```diff
@@ -0,0 +1,46 @@
+import json
+import logging
+
+from libcatapult.queues.base_queue import BaseQueue
+
+from workfinder import get_config
+from workfinder.api.espa_api import EspaAPI
+from workfinder.wait.basewaiter import BaseWaiter
+
+
+class Landsat8(BaseWaiter):
+
+    def __init__(self, q: BaseQueue, espa: EspaAPI):
+        super().__init__(q)
+        self.espa = espa
+
+    def check_order(self, order_id: str):
+        resp = self.espa.call(f'item-status/{order_id}', body={'status': 'complete'})
+        logging.info(f"item-status returned: {resp}")
+        return len(resp[order_id]) > 0
+
+    def send_complete_order(self, order_details: dict):
+        order_id = order_details['order_id']
+        resp = self.espa.call(f"item-status/{order_details['order_id']}", body={'status': 'complete'})
+        target_queue = get_config("landsat8", "redis_channel")
+        target_bucket = get_config("AWS", "bucket")
+        for item in resp[order_id]:
+            url = item.get('product_dload_url')
+            if 'LE07' in url:
+                nm = 'landsat_7'
+            elif 'LT05' in url:
+                nm = 'landsat_5'
+            elif 'LT04' in url:
+                nm = 'landsat_5'
+            elif 'LC08' in url:
+                nm = 'landsat_8'
+            else:
+                raise Exception("unknown landsat url")
```

Comment:
I always try to avoid these chained if statements, if I could suggest you something, I usually use this pattern for this situations, also helps to avoid possible typos like in line 34:

```suggestion
           landsat_registry = {
                'LE04': 'landsat_7',
                'LE05': 'landsat_5',
                'LE07': 'landsat_7',
                'LE08': 'landsat_8',
            }
            for k, v in landsat_registry.items():
                if k in url:
                    nm = v
            if not nm:
                raise Exception("unknown landsat url")
```

P.S: Ideally, landsat_registry will be part of a configuration file that is loaded/modified and can be extensible without touching any code

Reply:
Good spot! I'll admit I copied that over from the notebook version with out checking it too well.